timemachine/fe/mcgregor.py

```diff
@@ -135,45 +126,105 @@ def __init__(self, n_vertices, edges):
             for edge_idx in edges:
                 self.ve_matrix[vertex_idx][edge_idx] = True
 
-    def mapping_is_disconnected(self, source, node_states, n_mapped_nodes):
-        r"""
-        Verify whether mapped nodes can still be connected in the graph. It is assumed that
-        source is a mapped node. i.e. node_states[source] == NODE_STATE_VISITED_MAPPED. If this function
-        returns True, then the resulting graph is definitely disconnected. If this function
-        returns False, then the resulting graph can still be connected.
-
-        For example, let M be mapped node, D be a demapped node, and U be an unvisited node:
+    def mapping_is_disconnected(
+        self,
+        mapped_nodes: Set[int],
+        unvisited_nodes: Set[int],
+        max_connected_components: Optional[int],
+        min_connected_component_size: int,
+    ):
+        r"""Verify whether mapped nodes can still be connected at least the specified minimum number of connected
+        components. It is assumed that source is a mapped node. i.e. node_states[source] == NODE_STATE_VISITED_MAPPED.
```

Comment:
Stale documentation: I don't think we use node_states anymore.

Reply:
Updated this docstring (with other fixes as well) in https://github.com/proteneer/timemachine/pull/1349/commits/cfd7cbf9039337c52308d049473a324fad90a9d0. Also renamed the method to reflect its generality